### PR TITLE
Initial trip checks changes #2151

### PR DESCRIPTION
**Pull request by Mak-NOAA**
Initial changes for issue #2151:  https://github.com/nwfsc-fram/boatnet/issues/2151

### CODE REVIEW

**Review by melinashak-noaa — CHANGES_REQUESTED** [review]
Thanks for working on this Mak and getting everything setup. Just a few comments.

**Review by sethgerou-noaa — APPROVED** [review]
see comments on dropping errors docs

File: routes/index.ts [hunk 1]
```diff
@@ -564,6 +564,30 @@ const getDocs = async (req, res ) => {
 //     res.status(200).send('see console');
 // }
 
+
+const runTripChecks = async (req, res) => {
+    const trip = await masterDev.get(req.query.trip_id);
+    const previousTripErrors = await masterDev.view('obs_web', 'wcgop-trip-errors', {include_docs: true, reduce: false, key: trip.legacy.tripId});
+    
+    for (var previousTripError of previousTripErrors.rows) {
+        masterDev.destroy(previousTripError.doc._id, previousTripError.doc._rev);
```
[the remark this answers is not in the document]
Comment (melinashak-noaa) [reply routes/index.ts:573]: Got it, that makes sense, yes I like the idea of an errors array in the trips doc. Although would it get too large that maybe we need to store it in it's own doc? Well yes storing it in the trip doc should be good.

[the remark this answers is not in the document]
Comment (sethgerou-noaa) [reply routes/index.ts:573]: was thinking a separate errors doc with array of errors so as not to clutter the trip doc.

[the remark this answers is not in the document]
Comment (melinashak-noaa) [reply routes/index.ts:573]: sure that works

[the remark this answers is not in the document]
Comment (Mak-NOAA) [reply routes/index.ts:573]: @sethgerou-noaa I've setup some time for us to set up the trip errors as an array in trip record tomorrow

File: routes/index.ts [hunk 2]
```diff
@@ -564,6 +564,30 @@ const getDocs = async (req, res ) => {
 //     res.status(200).send('see console');
 // }
 
+
+const runTripChecks = async (req, res) => {
```
[the remark this answers is not in the document]
Comment (Mak-NOAA) [reply routes/index.ts:568]: @melinashak-noaa yeah we just quickly made that up....I like runTripErrorChecks

**Review by melinashak-noaa — APPROVED** [review]

File: routes/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mak-NOAA) [reply routes/index.ts]: We created a trip error array within the trip error document.

**Pull request merged**